nixos/modules/services/misc/etesync-dav.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.etesync-dav;
+in
+  {
+    options.services.etesync-dav = {
+      enable = mkEnableOption "etesync-dav";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.etesync-dav;
+        defaultText = literalExample "pkgs.etesync-dav";
+        description = "The package to use for the etesync-dav service.";
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "etesync-dav host";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 37358;
+        description = "etesync-dav port";
+      };
+
+      apiUrl = mkOption {
+        type = types.str;
+        default = "https://api.etesync.com/";
+        description = "The url to the etesync API";
+      };
+
+      openFirewall = mkOption {
+        default = false;
+        type = types.bool;
+        description = "Whether to open the firewall for the specified port.";
+      };
+    };
+
+    config = mkIf cfg.enable {
+      networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.port ];
+
+      systemd.services.etesync-dav = {
+        description = "etesync-dav - A CalDAV and CardDAV adapter for EteSync";
+        after = [ "network-online.target" ];
+        wantedBy = [ "multi-user.target" ];
+        path = [ cfg.package ];
+        environment = {
+          ETESYNC_LISTEN_ADDRESS = cfg.host;
+          ETESYNC_LISTEN_PORT = toString cfg.port;
+          ETESYNC_URL = cfg.apiUrl;
+          ETESYNC_DATA_DIR = "/var/lib/etesync-dav";
+        };
+
```

Comment:
There *might* be one or two more options you could from https://github.com/etesync/etesync-dav/blob/master/examples/systemd-sandbox/etesync-dav%40.service but I haven't verified that and could easily be wrong.

Reply:
I originally based that list on the [config.py](https://github.com/etesync/etesync-dav/blob/a90a1760fda7f38c8e432503adbc35c9335749b2/etesync_dav/config.py) file. 

`ETESYNC_CONFIG_DIR` is a legacy option and no longer necessary, and several of the other options could result in the files being outside of the `StateDirectory` which would also cause errors (e.g. `ETESYNC_DATABASE_FILE`, `CREDS_FILE`, etc).

As you mentioned in a previous comment, allowing SSL support would be good, so I think `SSL_CERT_FILE` and `SSL_KEY_FILE` should be added. I'll update the PR.

Reply:
I was actually referring to `systemd` options, but ssl options sound great too :+1:

Reply:
~~I think you need to add `SSL_CERT_FILE` and `SSL_KEY_FILE` to `environment`.~~ Scratch that. After looking at [config.py](https://github.com/etesync/etesync-dav/blob/a90a1760fda7f38c8e432503adbc35c9335749b2/etesync_dav/config.py) I notice the default value is set to that.